manticore/native/models.py

```diff
@@ -268,3 +269,59 @@ def strcpy(state: State, dst: Union[int, BitVec], src: Union[int, BitVec]) -> Un
     if "strcpy" in state.context:
         del state.context["strcpy"]
     return ret
+
+
+def strncpy(
+    state: State, dst: Union[int, BitVec], src: Union[int, BitVec], n: Union[int, BitVec]
+) -> Union[int, BitVec]:
+    """
+    strncpy symbolic model
+
+    Algorithm:  Copy n bytes from src to dst. If the length of the src string is less than n pad the difference
+    with NULL bytes. If a symbolic byte is found that can be NULL but is not definitely NULL fork and concretize states.
+
+    :param state: current program state
+    :param dst: destination string address
+    :param src: source string address
+    :param n: number of bytes to copy
+    :return: pointer to the dst
+    """
+
+    if issymbolic(src):
+        raise ConcretizeArgument(state.cpu, 2)
+    if issymbolic(dst):
+        raise ConcretizeArgument(state.cpu, 1)
```

Comment:
Nit: Let's reorder this
```suggestion
    if issymbolic(dst):
        raise ConcretizeArgument(state.cpu, 1)
    if issymbolic(src):
        raise ConcretizeArgument(state.cpu, 2)
```